tasks/install_fpga_packages.yml

```diff
@@ -0,0 +1,117 @@
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Fail if fpga_rsu_driver_pkg_url is undefined and empty
+  fail:
+    msg: "fpga_rsu_driver_pkg_url is a mandatory variable"
+  when:
+    - not (( fpga_rsu_driver_pkg_url is defined) and (fpga_rsu_driver_pkg_url | length > 0))
+
+- name: Create fpga driver package directory if not exist
+  file:
+    path: "{{ fpga_rsu_driver_dir }}"
+    state: directory
+    mode: 0755
+
+- name: Copy and untar FPGA driver tarball
+  block:
+    - name: Set fact for rpm tarball name
+      set_fact:
+        fpga_rsu_tarball_name: "{{ fpga_rsu_driver_pkg_url | basename }}"
+
+    - name: Copy fpga driver tarball
+      copy:
+        src: "{{ fpga_rsu_driver_pkg_url }}"
+        dest: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name }}"
+      failed_when: false
+
+    - name: untar fpga driver tarball
+      unarchive:
+        src: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name }}"
+        dest: "{{ fpga_rsu_driver_dir }}"
+      register: fpga_driver_tarball_untar
+      failed_when: false
+
+- name: Check and remove opae packages if already installed
+  block:
+    - name: Check whether opae package already installed or not
+      shell: |-
+        set -o pipefail
+        rpm -qa | grep opae
+      register: rpm_opae_installed_check
+      ignore_errors: true
+
+    - name: remove previous opae packages
+      shell:
+        rpm -e "{{ rpm_opae_installed_check.stdout_lines | join(' ') }}"
+      register: old_opae_packages_removed
+      when: rpm_opae_installed_check.rc == 0
+      ignore_errors: true
+
+- name: Install packages and verify installation
+  block:
+    - name: setting fact for fpga driver rpm directory
+      set_fact:
+        fpga_driver_rpm_dir: "{{ fpga_rsu_driver_dir }}/{{ fpga_rsu_tarball_name | splitext | first | splitext | first }}"
+
+    - name: Setting fact for fpga install shell script
+      shell: |-
+        set -o pipefail
+        ls "{{ fpga_driver_rpm_dir }}" | grep *.sh
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}"
+      register: fpga_driver_install_script
+
+    - name: Install fpga drivers
+      shell:
+        "{{ fpga_driver_install_script }}" -y
+      args:
+        chdir: "{{ fpga_driver_rpm_dir }}"
+      register: opae_packages_installed
+      when: old_opae_packages_removed.rc == 0
+
+    - name: Fail when OPAE FPGA package installation failed
+      fail:
+        msg: "OPAE FPGA package installation failed."
+      when: opae_packages_installed.rc != 0
+
+    - name: Verify opae and intel packages installed
+      shell: |-
+        set -o pipefail
+        rpm -qa | grep 'opae'
+      when: opae_packages_installed.rc == 0
+
+    - name: Verify opae driver installation
+      shell: |-
+        set -o pipefail
+        lsmod | grep fpga
+        lsmod | grep pac_n3000_net
+      register: verify_opae_driver
+
+- name: Verify module is loaded in kernel and update flash
```

Comment:
Looks good in general, a few comments:

1) Depending on the currently installed firmware version, the upgrade procedure is different.
see https://www.intel.com/content/www/us/en/programmable/documentation/xgz1560360700260.html#hpo1573151952874

2) There are a number of checks that should be performed before firmware upgrade. From the same document above:

Note: These upgrades erase the Static Region (SR) root entry hash and any CSK cancellation IDs previously programmed in the flash of the Intel® FPGA PAC N3000.
Remember:
    Stop any service or daemon accessing the FPGA or XL710 before updating the Intel® FPGA PAC N3000 such as fpgad.
    PLDM requests may return stale data. Avoid Host PLDM requests.
    Ensure cooling requirements are met. The server can reboot if the FPGA Core temperature exceeds 95°C. For more information, refer to Cooling Requirements.

Tip: Before you proceed with upgrade, ensure that the FPGA Die Temperature is below 80°C using the following command:

sudo fpgainfo bmc

If it is higher than the threshold value, increase the fan speed to improve thermal condition.

Reply:
Also, perhaps its good to only attempt to perform RPM installation if we know the packages work on RHEL-8.

The current packages work on RHEL-7 only.

Reply:
Thank you Marcelo for review.

We are only trying to install rpm in this and I will update the README to improve doc and update the pull-request.